RadixWallet/Clients/ProfileStore/ProfileStore.swift

```diff
@@ -288,364 +223,306 @@ extension ProfileStore {
+		guard profile.header.id == self.profile.header.id else {
+			let errorMessage = "Incorrect implementation: `\(#function)` was called with a Profile which UUID does not match the current one. This should never happen."
+			loggerGlobal.critical(.init(stringLiteral: errorMessage))
+			assertionFailure(errorMessage)
+			throw Error.profileIDMismatch
 		}
+		// All good
 	}
 
-	func assertDeviceOwnsSnapshotElseCreateNew(_ snapshot: ProfileSnapshot) async throws {
-		await Self.checkIfDeviceOwnsProfileSnapshot(snapshot)
-		// FIXME: Reintroduce later
-		//        do {
-//		} catch {
-//			// Note: We do not reset the active profile id, as doing so, will imply that user has no profile.
-//			//       Instead, we will prompt that the user, that the currently active profile is used on other device.
-//
-//			// Go to ephemeral state straightaway. The Wallet will redirect user to the Onboarding screen.
-//			await changeState(to: .ephemeral(.init(
-//				profile: Self.newEphemeralProfile(),
-//				loadFailure: .profileUsedOnAnotherDevice(error)
-//
-//			// rethrow the error to halt the execution up the chain
-//			throw error
-//		}
-	}
-
-	// The implementation of this is not what we want in the future, it is
-	// written like this to rectify bad state for some users who might incorrectly
-	// have a discprenacy between their `profile.header.lastUsedOnDevice.id` and
-	// the `deviceIdentifer` saved in keychain
-	static func checkIfDeviceOwnsProfileSnapshot(_ profileSnapshot: ProfileSnapshot) async {
-		@Dependency(\.secureStorageClient) var secureStorageClient
-
-		// Load the last used device info
-		let lastUsedOnDevice = profileSnapshot.header.lastUsedOnDevice
+	private func _assertOwnership() throws {
+		try _assertOwnership(of: profile)
+	}
 
-		if let deviceID = try? await secureStorageClient.loadDeviceIdentifier() {
-			if lastUsedOnDevice.id != deviceID {
-				loggerGlobal.notice("DeviceIdentifier discrepancy, in profile header \(lastUsedOnDevice.id) != \(deviceID) (loaded from keychain) => rectifiying by saving the one found in profile header into keychain.")
-				do {
-					try await secureStorageClient.saveDeviceIdentifier(lastUsedOnDevice.id)
-				} catch {
-					loggerGlobal.error("Failed to rectify deviceID discrepancy for users, error: \(error) (mismatch)")
-				}
-			} else {
-				loggerGlobal.trace("Verified that device owns profile snapshot ✅")
-			}
-		} else {
-			loggerGlobal.notice("Found no DeviceIdentifier in keychain rectifiying by saving the one found in profile header: \(lastUsedOnDevice.id) into keychain.")
-			do {
-				try await secureStorageClient.saveDeviceIdentifier(lastUsedOnDevice.id)
-			} catch {
-				loggerGlobal.error("Failed to rectify deviceID discrepancy for users, error: \(error) (failed to load deviceID from keychain)")
+	private func _assertOwnership(of profile: Profile) throws {
+		try Self._assertOwnership(of: profile, against: deviceInfo) {
+			Task {
```

Comment:
This task could be inside `emitOwnershipConflict`, which then will become a sync nonThrowing function.

Reply:
sure! does not help us in `init` though :/